netbox/users/views.py

```diff
@@ -46,9 +46,18 @@ def get(self, request):
             logger = logging.getLogger('netbox.auth.login')
             return self.redirect_to_next(request, logger)
 
-        auth_backends = {
-            name: get_auth_backend_display(name) for name in load_backends(settings.AUTHENTICATION_BACKENDS).keys()
-        }
+        auth_backends = {}
+        for name in load_backends(settings.AUTHENTICATION_BACKENDS).keys():
```

Comment:
Let's refactor this to add a new backend to `auth_backends` for each IDP. Then we can just iterate through the backends list in the template without the additional complexity of checking for `saml_idps`.

It's probably also worth rewriting `get_auth_backend_display()` to just return the dictionary of display name and icon directly.